lib/screens/transfer/send/send_enter_data/interactor/mappers/send_amount_change_mapper.dart

```diff
@@ -12,17 +13,19 @@ class SendAmountChangeMapper extends StateMapper {
     final tokenAmount = TokenDataModel(parsedQuantity, token: settingsStorage.selectedToken);
     final selectedFiat = settingsStorage.selectedFiatCurrency;
     final fiatAmount = rateState.tokenToFiat(tokenAmount, selectedFiat);
+    final toAccount = currentState.sendTo.account;
 
     final double currentAvailable = currentState.availableBalance?.amount ?? 0;
 
-    final bool enoughBalance = parsedQuantity <= currentAvailable ||
-        currentState.availableBalance?.asFormattedString() == tokenAmount.asFormattedString();
+    final double insufficiency =
+        currentState.availableBalance?.asFormattedString() == tokenAmount.asFormattedString() ? 0.0 :
+        parsedQuantity - currentAvailable;
 
     return currentState.copyWith(
       fiatAmount: fiatAmount,
-      isNextButtonEnabled: parsedQuantity > 0 && enoughBalance,
```

Comment:
Passing around "insufficiency" breaks my brain.... 😄

Reply:
Agree this is messy.
The challenge comes when you start using token contracts that are smarter than eosio.token -- e.g. Rainbows. The token contract has internal logic, and ideally you would like to mimic that logic in the wallet app so you can speak nicely to the user rather than give them a failed transaction on the blockchain.
This makes the wallet logic tightly coupled to the token smart contract, which is not great if you want to allow evolution in the token contract space.
Works now, refactor later?

Reply:
instead of blocktransfer, should be allowTransfer(amount, availableAmount

Fix in another PR, agreed